lib/filedrop.ts

```diff
@@ -23,14 +25,33 @@ function firstMatchingItem(list: DataTransferItemList, acceptVal: string): DataT
       }
     }
     return false;
-  });
+  };
+
+  let results: DataTransferItem[];
+
+  if (multiple) {
+    results = dataItems.filter(predicate);
+  } else {
+    const item = dataItems.find(predicate);
+    if (item == null) return [];
+    results = [item];
+  }
+
+  return results;
 }
 
-function getFileData(data: DataTransfer, accept: string): File | undefined {
-  const dragDataItem = firstMatchingItem(data.items, accept);
-  if (!dragDataItem) return;
+function getFileData(data: DataTransfer, accept: string, multiple: boolean): File[] {
+  const dragDataItems = getMatchingItems(data.items, accept, multiple);
+  const files: File[] = [];
+
+  // This is because Map doesn't like the null type returned by getAsFile
```

Comment:
That’s interesting. What exactly happens? Would a subsequent `filter()` work?

Reply:
I chose not to change this.